core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceTracker.java

```diff
@@ -381,64 +364,61 @@ private Resource getResource(URL location) {
         }
     }
 
-    /**
-     * Wait for some resources.
-     *
-     * @param resources the resources to wait for
-     * @throws InterruptedException if another thread interrupted the wait
-     */
-    private void wait(Resource... resources) throws InterruptedException {
-        // save futures in list to allow parallel start of all resources
-        final List<Future<Resource>> futures = Stream.of(resources)
-                .map(ResourceHandler::new)
-                .map(ResourceHandler::putIntoCache)
-                .collect(Collectors.toList());
-
-        for (Future<Resource> future : futures) {
-            try {
-                future.get();
-            } catch (ExecutionException ignored) {
-            }
-        }
+    private Future<Resource> triggerDownloadFor(Resource resource, final Executor downloadExecutor) {
```

Comment:
please move this method one down (after `waitForCompletion(Resource... resources)`)